scenarios/test_codegen/test/LoadLayer_test.res

```diff
@@ -0,0 +1,443 @@
+      // but we are not sure that it'll bring some benefits
+      Assert.deepEqual(
+        mock.loadEntitiesByIdsCalls,
+        [
+          {
+            entityIds: ["2"],
+            entityMod: module(Entities.User)->Entities.entityModToInternal,
+            logger: Logging.logger,
+          },
+          {
+            entityIds: ["3"],
+            entityMod: module(Entities.User)->Entities.entityModToInternal,
+            logger: Logging.logger,
+          },
+        ],
+      )
+      Assert.deepEqual(mock.loadEntitiesByFieldCalls, [])
+    },
+  )
+
+  Async.it("Trys to load non existing entities from db by field", async () => {
+    let mock = Mock.LoadLayer.make()
+
+    let getUsersWithId =
+      mock.loadLayer->LoadLayer.makeWhereEqLoader(
+        ~entityMod=module(Entities.User),
+        ~inMemoryStore=InMemoryStore.make(),
+        ~logger=Logging.logger,
+        ~fieldName="id",
+        ~fieldValueSchema=S.string,
+      )
+
+    let users = await getUsersWithId("123")
+
+    Assert.deepEqual(users, [])
+    Assert.deepEqual(mock.loadEntitiesByIdsCalls, [])
+    Assert.deepEqual(
+      mock.loadEntitiesByFieldCalls,
+      [
+        {
+          fieldName: "id",
+          fieldValue: "123"->Utils.magic,
+          fieldValueSchema: S.string->Utils.magic,
+          entityMod: module(Entities.User)->Entities.entityModToInternal,
+          logger: Logging.logger,
+        },
+      ],
+    )
+  })
+
+  Async.it("Gets entity from inMemoryStore by index if it exists", async () => {
+    let mock = Mock.LoadLayer.make()
+
+    let user1 = (
+      {
+        id: "1",
+        accountType: USER,
+        address: "",
+        gravatar_id: None,
+        updatesCountOnUserForTesting: 0,
+      }: Entities.User.t
+    )
+
+    let inMemoryStore = Mock.InMemoryStore.make(~entities=[(module(Entities.User), [user1])])
+
+    let getUsersWithId =
+      mock.loadLayer->LoadLayer.makeWhereEqLoader(
+        ~entityMod=module(Entities.User),
+        ~inMemoryStore,
+        ~logger=Logging.logger,
+        ~fieldName="id",
+        ~fieldValueSchema=S.string,
+      )
+
+    let users = await getUsersWithId("1")
+
+    // FIXME: Should register indexes for entities set in the inMemoryStore before the index creation
+    // Assert.deepEqual(users, [user1])
+    // Assert.deepEqual(mock.loadEntitiesByIdsCalls, [])
+    // Assert.deepEqual(mock.loadEntitiesByFieldCalls, [])
+
+    Assert.deepEqual(users, [])
+    Assert.deepEqual(mock.loadEntitiesByIdsCalls, [])
+    Assert.deepEqual(
+      mock.loadEntitiesByFieldCalls,
+      [
+        {
+          fieldName: "id",
+          fieldValue: "1"->Utils.magic,
+          fieldValueSchema: S.string->Utils.magic,
+          entityMod: module(Entities.User)->Entities.entityModToInternal,
+          logger: Logging.logger,
+        },
+      ],
+    )
+  })
+
+  Async.it(
+    "Correctly gets entity from inMemoryStore by index if the entity set after the index creation",
```

Comment:
https://github.com/enviodev/hyperindex/pull/105 This should change the outcome of this test right?

Reply:
Yes